react-training/src/redux/auth/action.js

```diff
@@ -4,39 +4,43 @@ import { getByEmail } from '../../services/loginService';
 export const actionTypes = {
   SET_AUTHENTICATION: 'SET_AUTHENTICATION',
   LOGIN_REQUEST: 'LOGIN_REQUEST',
-  LOGIN_SUCCESS: 'LOGIN_SUCCESS',
-  LOGIN_FAILURE: 'LOGIN_FAILURE',
+  LOGIN_REQUEST_SUCCESS: 'LOGIN_SUCCESS',
+  LOGIN_REQUEST_FAILURE: 'LOGIN_FAILURE',
   LOG_OUT: 'LOG_OUT'
 };
 
+const TARGET = 'auth';
+
 const authActions = {
-  setAuthentication: () => dispatch => {
+  setAuthentication: () => {
     const sessionUser = sessionExists();
-    const payload = {};
+    let payload = null;
     if (sessionUser) {
-      payload.user = sessionUser;
-      payload.isAuth = true;
-      dispatch({ type: actionTypes.SET_AUTHENTICATION, payload });
+      payload = true;
     }
+    return { type: actionTypes.SET_AUTHENTICATION, payload, target: TARGET };
   },
   requestLogin: login => async dispatch => {
-    dispatch({ type: actionTypes.LOGIN_REQUEST, login });
-    /* mock server latency */
+    dispatch({
+      type: actionTypes.LOGIN_REQUEST,
+      target: TARGET
+    });
+
     setTimeout(async () => {
-      const apiResponse = await getByEmail(login);
+      const apiResponse = await getByEmail(login.email);
```

Comment:
Let's use fetch middleware here. As there is no documentation of this, when the time comes to implement it, you let me know so I explain it to you in person